docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -52,52 +53,130 @@ const theme = createTheme({
 
 {{"demo": "GlobalThemeOverride.js"}}
 
-Each component is composed of several different parts.
-These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
-You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
+### Variants
+
+Most components support props that exist to change their design.
+For example, the Card component supports a `variant` prop where you can pick `outlined` as a value that adds a border.
+
+If you want to override styles from a specific prop, create `variants` of the specific slot that contains `props` and `style` keys. When the component's `props` matches, the `style` will be applied.
+
+The override definition must specified in an array.
+Also, ensure that any styles that should take precedence are listed last.
+
+**Example 1**: Overriding styles based on existing props. To increase the border thickness of the `outlined` Card, use this following snippet:
```

Comment:
I think we could use regular headings and remove "Example X", which is not very informative. It'd probably require reword some of the paragraphs.

```suggestion
#### Overriding styles based on existing props

To increase the border thickness of the `outlined` Card, use the following snippet:
```

Reply:
Updated the rest as well, thanks.